### ISSUE TITLE
Can't run `bunx supabase` cli executable

### ISSUE THREAD

**Issue by cpreston321** [oven-sh/bun#4583]
### What version of Bun is running?

1.0.0+822a00c4d508b54f650933a73ca5f4a3af9a7983

### What platform is your computer?

Darwin 22.6.0 arm64 arm

### What steps can reproduce the bug?

Run `bunx supabase` 

### What is the expected behavior?

Runs the supabase cli. You can get this to work using `npx supabase` and will run like normal

### What do you see instead?

```
error: could not determine executable to run for package supabase
```

### Additional information

I've tried installing supabase globally using `bun i -g supabase` it installs but runs into the same error.

**Comment by apollo79** [comment]
I'm facing this issue too.

**Comment by lorenzorapetti** [comment]
`supabase` uses a postinstall script which [bun doesn't run by default](https://bun.sh/docs/cli/install#trusted-dependencies), so you have to add this to the `package.json`:

```json
"trustedDependencies": [
  "supabase"
]
```

I also had to delete `node_modules` and `bun.lockb`

**Comment by uncenteredDiv** [comment]
@lorenzorapetti
I tried the way you described - but it unfortunately still doesn't work 🤔 

I deleted `node_modules` and `bun.lockb` - removed / installed the `supabase` again - run `bun i` - etc.

**Comment by lorenzorapetti** [comment]
You can try again deleting bun's cache: `bun pm cache rm` (you also have to delete `node_modules` and `bun.lockb` again)

**Comment by uncenteredDiv** [comment]
@lorenzorapetti that worked! Thanks 🙏

**Comment by zachbryant** [comment]
There's a broader issue of binary files that aren't from postInstall not showing up, so for some packages this CLI bug remains. In fact, my entire node_modules is full of empty folders (no package.json, lib folders, js files, etc)

**Comment by colinhacks** [comment]
Duplicate of https://github.com/oven-sh/bun/issues/4959

**Comment by ozanvos** [comment]
I think the tips above do not work if the package is not installed in your project. I have a project which is calling `npx supabase@latest` in a script without having it in devDependencies. When I run my script Bun is rewriting `npx` to `bunx`. Looking at the command help, I don't see a way to pass trustedDependencies to bunx.

This is easy enough to work around by using `npm run` instead, but a paper cut nevertheless.

**Comment by silentworks** [comment]
@osener true but you should really be locking a version of the supabase cli into your project otherwise you could call `npx supabase@latest start` and your project breaks because something has changed in the latest version that was released.

**Comment by osener** [comment]
> @osener true but you should really be locking a version of the supabase cli into your project otherwise you could call `npx supabase@latest start` and your project breaks because something has changed in the latest version that was released.

I've used to do that in the past but removed it from my project dependencies as it was causing problems during Cloudflare CI build. This may have been fixed now but I've come to prefer using `npx supabase@latest start` as a standalone tool.

I am aware of the risks. However, in my experience my local dev environment being out of date while the production cloud  Supabase Console always changing causes more issues / degraded experience than not pinning my cli version. I prefer always having an up to date local environment that is closer to production.

Regarding new versions breaking my project, I never experienced that except one time when the CLI was not working at all for a short time, but I was able to work around that by using a specific version. But this was fixed promptly so I went back to `@latest`.

Sure, there might come a time when the CLI and/or Supabase local dev environment changes so much that it would break if I upgrade the CLI. In my opinion this should be handled gracefully with a fair bit of warning. Maybe a Supabase major version field should be introduced to `config.toml` so that some backwards compatibility can be maintained, and users can be warned that they'll have to migrate their development setup with ample amount of time. This can be like how Remix did their major version upgrade, or how Cloudflare has `compatibility_date` in their wrangler.toml files. If all of a sudden updating of the CLI (whether in my project dependencies or when using `@latest`) breaks my dev environment, I would also be very worried about my production breaking unexpectedly anyway.

**Comment by ozanvos** [comment]
> @osener true but you should really be locking a version of the supabase cli into your project otherwise you could call `npx supabase@latest start` and your project breaks because something has changed in the latest version that was released.

@silentworks thanks for the suggestion. I've used to do that in the past but removed it from my project dependencies as it was causing problems during Cloudflare CI build. This may have been fixed now but I've come to prefer using `npx supabase@latest start` as a standalone tool.

I am aware of the risks. However, in my experience my local dev environment being out of date while the production cloud  Supabase Console always changing causes more issues / degraded experience than not pinning my cli version. I prefer always having an up to date local environment that is closer to production.

Regarding new versions breaking my project, I never experienced that except one time when the CLI was not working at all for a short time, but I was able to work around that by using a specific version. But this was fixed promptly so I went back to `@latest`.

Sure, there might come a time when the CLI and/or Supabase local dev environment changes so much that it would break if I upgrade the CLI. In my opinion this should be handled gracefully with a fair bit of warning. Maybe a Supabase major version field should be introduced to `config.toml` so that some backwards compatibility can be maintained, and users can be warned that they'll have to migrate their development setup with ample amount of time. This can be like how Remix did their major version upgrade, or how Cloudflare has `compatibility_date` in their wrangler.toml files. If all of a sudden updating of the CLI (whether in my project dependencies or when using `@latest`) breaks my dev environment, I would also be very worried about my production breaking unexpectedly anyway.
